src/SFML/Graphics/SpriteBatch.cpp

```diff
@@ -0,0 +1,266 @@
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Graphics/Font.hpp>
+#include <SFML/Graphics/RenderStates.hpp>
+#include <SFML/Graphics/RenderTarget.hpp>
+#include <SFML/Graphics/SpriteBatch.hpp>
+#include <SFML/System/Err.hpp>
+
+#include <algorithm>
+
+namespace sf
+{
+////////////////////////////////////////////////////////////
+void SpriteBatch::begin(BatchMode batchMode, RenderTarget& renderTarget, const Shader* shader, const BlendMode& blendMode)
+{
+    if (m_batchActive)
+    {
+        err() << "A batch is already active.";
+        return;
+    }
+
+    m_batchMode    = batchMode;
+    m_blendMode    = blendMode;
+    m_shader       = shader;
+    m_transform    = Transform::Identity;
+    m_renderTarget = &renderTarget;
+    m_batchActive  = true;
+}
+
+
+////////////////////////////////////////////////////////////
+void SpriteBatch::draw(const Batchable& drawable, float depth)
+{
+    drawable.batch(*this, depth);
+}
+
+
+////////////////////////////////////////////////////////////
+void SpriteBatch::draw(const Vertex*    vertices,
+                       std::size_t      count,
+                       PrimitiveType    type,
+                       const Texture*   texture,
+                       const Transform& transform,
+                       float            depth)
+{
+    bool unsupportedType = type != PrimitiveType::TriangleFan && type != PrimitiveType::TriangleStrip &&
+                           type != PrimitiveType::Triangles;
+
+    if (unsupportedType)
+    {
+        err() << "SpriteBatch supports only triangle-based primitive types.";
+        return;
+    }
+
+    if (count == 0)
+        return;
+
+    if (m_batchMode == BatchMode::Immediate)
+    {
+        // Immediate is the same as using a RenderTarget directly,
+        // so we don't use VertexBuffer here
+        m_renderTarget->draw(vertices, count, type, {m_blendMode, m_transform * transform, texture, m_shader});
+        return;
+    }
+
+    // Split into triangles if needed
+
+    if (type == PrimitiveType::TriangleStrip)
```

Comment:
Would this make more sense as a switch case than `if`/`else if` blocks?

Reply:
The `switch` version looks a bit more cluttered to me.

Reply:
Using `switch` is more expressive and means that if a new enum is added, the compiler will be sure to emit a warning about not handling that new enum. It makes the code easier to understand and easier for others to modify months or years from now. If the language already offers a tailor-made solution, we ought to use it.